File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    # brown = yellow로 만들수 있는 가로 * 세로 + 4
    for i in range(1, int(yellow**0.5)+1):
        width = yellow // i
        height = i
        if (width+height) == (brown-4)//2:
            answer = [width+2, height+2]
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    for (let col = 1; col <= yellow**0.5; col++) {
        let row = yellow / col;
        if ((row*2 + col*2 + 4) === brown) {
            answer.push(row+2);
            answer.push(col+2);
            break;
        }        
    }
    return answer;
}